### PR TITLE
Migrate:grade task modal

### PR DESCRIPTION
**Pull request by leocomsci**
# Description

Transfer PR started by @PerryRose 

This pull request migrates the grade-task-modal component from AngularJS to Angular.
The grade task modal component is opened from task-service.coffee using the grade task modal service, which receives a task object and passes it onto the component via a MatDialogRef. Depending on the task object, a student's submission can either be rated using a Mat Slider, or graded with a Mat Button Toggle Group, which uses the grade-icon component which I have migrated. Clicking the submit button calls the component's close function, which passes a rating and a grade value back to the service before then being passed back to task-service.coffee, which updates the task object on the backend.

### Before:
image

### After:
image

### Before:
image

### After:
image

Fixes # (updates the old grade-task-modal component specified outlined in the migration task list)

## Type of change

_Please delete options that are not relevant._

- [x] Component migration

# How Has This Been Tested?

1. Log in as an admin
2. Go to Manage Units and choose a unit
4. Go to Tasks and choose a task
5. Go to Other Settings
6. To grade a task, tick the Graded option.
7. To rate a task, select a number for the Quality Points option
8. Save
9. Log in as a tutor
10. Go to Mark by Task Definition
11. Select a task
12. Select a submission and provide a rating

## Testing Checklist:

- [X] Tested in latest Chrome
- [X] Tested in latest Safari
- [X] Tested in latest Firefox

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have requested a review from _ and _ on the Pull Request

### CODE REVIEW

**Comment by leocomsci** [comment]
@macite Hi Andrew, could you review this PR? I can't add you as a reviewer for this PR so I mentioned you in this comment.
I also change name of welcome component, Jake used WelcomeWizardComponent but there's just WelcomeComponent so I adjusted it. Also, there are some problems with karma module so I couldn't run unit tests.

File: src/app/common/grade-icon/grade-icon.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  styleUrls: ['grade-icon.component.scss'],
+})
+export class GradeIconComponent implements OnInit {
+  @Input() grade: string = 'F';
+  @Input() index: number;
+
+  gradeText: string;
+  gradeLetter: string;
+
+  constructor(@Inject(gradeService) private GradeService: any) {}
```
Comment (macite) [inline src/app/common/grade-icon/grade-icon.component.ts]: There should be a new equivalent of this in typescript.

Comment (macite) [reply src/app/common/grade-icon/grade-icon.component.ts]: Check if you can remove the GradeService from this and use the new code

File: src/app/groups/group-member-contribution-assigner/group-member-contribution-assigner.tpl.html [hunk 2]
```diff
@@ -8,29 +8,29 @@
       </tr>
     </thead>
     <tbody>
-      <tr ng-repeat="contrib in team.memberContributions | orderBy:memberSortOrder:reverse">
-        <td>{{contrib.project.student.name}}</td>
+      <tr ng-repeat="member in team.members | orderBy:memberSortOrder:reverse">
+        <td>{{member.student_name}}</td>
```
Comment (macite) [inline src/app/groups/group-member-contribution-assigner/group-member-contribution-assigner.tpl.html:12]: this doesn't look right - with the new service we do not have any snake case properties

Comment (macite) [reply src/app/groups/group-member-contribution-assigner/group-member-contribution-assigner.tpl.html:12]: Check everything in this...

File: src/app/projects/states/portfolio/directives/portfolio-grade-select-step/portfolio-grade-select-step.tpl.html [hunk 3]
```diff
@@ -33,10 +33,7 @@ <h4>Grade Application</h4>
             ng-model="project.submitted_grade"
             btn-radio="{{$index}}"
           >
-            <grade-icon
-              grade="grade"
-              class="text-{{project.submitted_grade == grades.indexOf(grade) ? 'success' : 'muted'}}"
-            ></grade-icon>
+            <grade-icon [grade]="grade"></grade-icon>
```
Comment (macite) [inline src/app/projects/states/portfolio/directives/portfolio-grade-select-step/portfolio-grade-select-step.tpl.html]: Can you check the muted state here... is this ok without it. It is not shown in the conversation.

File: src/app/units/states/portfolios/portfolios.tpl.html [hunk 4]
```diff
@@ -43,7 +43,6 @@ <h4 class="panel-title">Mark Portfolios</h4>
               <span
                 class="grade-icon"
                 tooltip="Show all portfolios"
-                class="text-{{$index == filterOptions.selectedGrade ? 'primary' : 'muted'}}"
                 >All</span
```
Comment (macite) [inline src/app/units/states/portfolios/portfolios.tpl.html]: How does this appear now when different grades are selected?

File: src/app/units/states/portfolios/portfolios.tpl.html [hunk 5]
```diff
@@ -56,10 +55,9 @@ <h4 class="panel-title">Mark Portfolios</h4>
               btn-radio="{{$index}}"
             >
               <grade-icon
-                grade="grade"
+                [grade]="grade"
                 tooltip="Select to show student's aiming for {{grade}}"
                 tooltip-append-to-body="true"
-                class="text-{{$index == filterOptions.selectedGrade ? 'primary' : 'muted'}}"
```
Comment (macite) [inline src/app/units/states/portfolios/portfolios.tpl.html:71]: As above

**Review by macite — CHANGES_REQUESTED** [review]
This looks pretty close. Can you check these last few things and make adjustments or show that the implementation should be ok?

**Pull request merged**